### DISCUSSION TITLE
How to Access Cloudflare ExecutionContext in actions

### DISCUSSION THREAD

**Discussion by chrizy** [remix-run/remix#7019] [Q&A]
Hi, running Remix on Cloudflare workers, and want to take advantage of Cloudflare ExecutionContext.waitUntil

how can I access this from within a action.   thanks

```
export const action = async ({ request, context }: ActionArgs) => {
  ctx.waitUntil( promise ) // waitUntil allows processing like logging to happen after the response is returned
```

**Comment by sergiodxa** [comment]
The default template for Cloudflare Worker only sends the `env` key, but you can customize your `server.ts` file (the worker entry file) to pass `waitUntil` or the whole ExecutionContext object.

This is where that's configured https://github.com/remix-run/remix/blob/a8924777a8b8cefd02a58b0c9333bccefb0ffdc2/templates/cloudflare-workers/server.ts#L44-L46
